api/src/services/generate-xform.js

```diff
@@ -45,25 +45,30 @@ const transform = (formXml, stylesheet) => {
   });
 };
 
+const replaceLast = (haystack, needle) => {
+  const index = haystack.lastIndexOf(needle);
+  return haystack.slice(0, index) + haystack.slice(index + needle.length);
+};
+
+const replaceRootNode = (string, node) => {
+  return replaceLast(string.replace(node, ''), ROOT_CLOSE);
```

Comment:
Not totally versed in the schema: but should we only remove the first occurrence of `node`?

Reply:
When the first param is a string then replace only replaces the first occurrence. If it's a regex with `/g` then it replaces all. It's totally not clear here but it is covered by the new unit test.